src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/SexGenotypeData.java

```diff
@@ -0,0 +1,120 @@
+package org.broadinstitute.hellbender.tools.exome.sexgenotyper;
+
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.*;
+import java.util.stream.IntStream;
+
+/**
+ * A class for handling sex genotypes and their likelihoods.
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class SexGenotypeData {
+    /* mandatory */
+    private final String sampleName;
+    private final String sexGenotype;
+    private final boolean hasExtendedGenotypingInfo;
+
+    /* optional */
+    private final Map<String, Double> logLikelihoodMap;
```

Comment:
It is not the main component! one creates a `SexGenotypeData` record e.g. by reading a table. The table could have only the two mandatory columns (SAMPLE_NAME, SAMPLE_SEX_GENOTYPE) if e.g. it is provided by a user manually; or it could have additional columns for the likelihood of each sex genotype if it is inferred by our tool. What we _really_ need at the end of the day is the sex genotype of each sample. The likelihoods are extra info.